build-tools-internal/src/main/groovy/elasticsearch.auto-backporting.gradle

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+
+import org.elasticsearch.gradle.internal.rewrite.RewritePlugin
+
+allprojects {
+    apply plugin: 'elasticsearch.rewrite'
+    rewrite {
+        rewriteVersion = "7.10.0"
+        activeRecipe("org.elasticsearch.java.backport.ListOfBackport",
+                "org.elasticsearch.java.backport.MapOfBackport",
+                "org.elasticsearch.java.backport.SetOfBackport")
+        configFile = rootProject.file("rewrite.yml")
+    }
+
+    repositories {
+        mavenCentral()
+    }
+
+    configurations {
+        rewrite {
+            resolutionStrategy {
+                force 'org.jetbrains:annotations:21.0.1'
+                force 'org.slf4j:slf4j-api:1.7.31'
+                force 'org.jetbrains.kotlin:kotlin-stdlib:1.5.10'
+                force 'org.jetbrains.kotlin:kotlin-stdlib-common:1.5.10'
+            }
+        }
+    }
+
+    dependencies {
+        rewrite "org.openrewrite:rewrite-java-11"
+    }
+
+    if (gradle.getStartParameter().getTaskNames().any { it.endsWith(RewritePlugin.REWRITE_TASKNAME) }) {
+        afterEvaluate {
+            def java = project.getExtensions().findByType(JavaPluginExtension.class);
+            if (java) {
```

Comment:
the way the backporting works is that first rewrite creates the java11 based AST and then applys the rules on it to make it java8 compatible. if we don't have that building the AST fails as the java 11 methods are not resolvable in the first place

Reply:
I think I found a generic implementation for a rule to be configurable that allows us to avoid renaming and rely on full qualified names. need a bit more digging though to incorporate this

Reply:
Ah, interesting. I initially thought the AST was purely source-based and didn't rely on stuff being actually "compilable" but that makes sense.